casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala

```diff
@@ -97,15 +100,24 @@ object Genesis {
       walletsFile <- toFile[F](maybeWalletsPath, genesisPath.resolve("wallets.txt"))
       wallets     <- getWallets[F](walletsFile, maybeWalletsPath)
       bonds       <- getBonds[F](bondsFile, numValidators, genesisPath)
-      validators  = bonds.toSeq.map(Validator.tupled)
       genesisBlock <- createGenesisBlock(
                        runtimeManager,
                        Genesis(
                          shardId = shardId,
                          timestamp = timestamp,
                          wallets = wallets,
-                         proofOfStake = ProofOfStake(minimumBond, maximumBond, validators),
-                         faucet = faucet
+                         proofOfStake = ProofOfStake(
+    minimumBond = minimumBond,
+    maximumBond = maximumBond,
+    validators = bonds.toSeq.map(Validator.tupled)
+  ),
+                         faucet = faucet,
+                         genesisPk = Ed25519.newKeyPair._2,
+                         vaults = Traverse[List]
+    .traverse(bonds.keys.toList)(RevAddress.fromPublicKey)
```

Comment:
Rather than using  `bonds` twice in the`fromInputFiles` method, we should be passing the bonds (or Validator-s, doesn't matter) verbatim into `Genesis`, and doing this inside. For any production or test run, we want the validators initial vaults to have sufficient rev for them to bond.

Reply:
I'm not sure I understand. You mean just have `Seq[Validator]` or `Map[PublicKey, Long]` into the `Genesis` constructor, and handle the construction of `Seq[Vault]` in `Genesis`'s `apply`  method?

Reply:
in `createGenesis` method. `Genesis.apply is just for passing the parameters, no logic there.

Reply:
We shouldn't move the vault construction into `createGenesis` just like we shouldn't move wallet construction into `createGenesis`.

Reply:
Then we should not create vaults based on bonds. Instead, we should validate (inside createGenesis) that there are vaults corresponding to each bond, each with a sufficient amount.

Honestly though I'd move both Vault and Validator creation from `fromInputFiles` into `createGenesis`

Also, the validation that I asked you to include instead of Option.get should really be inside of `createGenesis`. We want valid behavior for all entrypoints, not just `fromInputFiles`. That's another reason for the paragraph above.